org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/packageview/PackageExplorerContentProvider.java

```diff
@@ -127,6 +128,18 @@ protected Object getViewerInput() {
 
 	@Override
 	public void elementChanged(final ElementChangedEvent event) {
+		IJavaElementDelta delta= event.getDelta();
+		if (Display.findDisplay(Thread.currentThread()) == null) {
+			processDelta(delta);
+		} else {
+			Job.create(PackagesMessages.PackageExplorerContentProvider_update_job_description, m -> {
```

Comment:
Should this be scheduled as system job so that the user is not notified of this update?

Reply:
Maybe yes ...